neuro_san/service/agent_server.py

```diff
@@ -116,11 +127,14 @@ def serve(self):
         # New-style service
         security_cfg = None     # ... yet
 
-        for agent_name, tool_registry in self.tool_registries.items():
+        self.setup_tool_factory_provider()
+        tool_factory_provider: ServiceToolFactoryProvider = \
+            ServiceToolFactoryProvider.get_instance()
 
+        for agent_name in self.tool_registries.keys():
```

Comment:
Shouldn't the keys come from the provider as well?

Reply:
Yes, that's a good point. Added a method to get a (snapshot) of current agent names.